contrib/vdsm-dev

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+# Author: Albert Esteve <aesteve@redhat.com>
+
+venv_path=$HOME/.venv/vdsm
+venv_activate_path=${venv_path}/bin/activate
+user_storage_path=/var/tmp/vdsm-storage/
+
+print_help()
+{
+    echo "oVirt/vdsm command line helper."
+    echo "Should be run from the repository root folder."
+    echo "Usage: $0 CMD [options]"
+    echo
+    echo "Available commands:"
+    echo "help:          Print this message and exits."
+    echo "build:         Build vdsm and make RPMs."
+    echo "build-install: Build vdsm, make RPMs, and update local packages."
+    echo "test:          Run any test from specified environment using tox."
+    echo "test-storage:  Run one/all storage tests using tox."
+    echo "show-coverage: Show coverage results generated after a test run."
+    echo "               Receives an environment with '-e env'. [default=storage]"
+}
+
+venv_create()
+{
+    echo "Creating venv at $venv_path"
+    python3 -m venv ${venv_path}
+    source "${venv_activate_path}"
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+}
+
+make_storage()
+{
+    if [ ! -d $user_storage_path ]; then
+        echo "Make storage at $user_storage_path"
+        make storage
+    fi
+}
+
+test_path()
+{
+    path=$1
+    storage_prefix=storage/
+    if [ ! -f "tests/$path" ]; then
+        path=${storage_prefix}${path}
+        if [ ! -f "tests/$path" ]; then
+            echo "No test $path found"
+            exit 1
+        fi
+    fi
+}
+
+vdsm_build()
+{
+    git clean -dxf && ./autogen.sh --system --enable-timestamp && \
+         make && rm -rf ~/rpmbuild && make rpm
+}
+
+handle_test_opts()
+{
+    while test $# -gt 0
+    do
+        case "$1" in
+            -e)
+                shift
+                env=$1
+                if [ $env == "storage" ]; then
+                    make_storage
+                fi
+                ;;
+            *)
+                # Let tox handle the rest of the arguments
+                ;;
+        esac
+        shift
+    done
+}
+
+
+if [ -z "$1" ]; then
+    echo "Command missing"
+    print_help
+    exit 1
+fi
+
+case $1 in
+
+    help )
+
+        print_help
+
+        ;;
+
+    build )
+
+        vdsm_build
```

Comment:
I would also suggest offering an option to build it in a podman container.  While the whole script can be run there, it's not obvious, according to my recent experience, for the general audience how to make it all properly.  A possibility to build Vdsm rpm's to a local ~/rpmbuild directory with a single command in any environment would make things much easier for those who need to run a particular version.  Also, running the build in a container would ensure the build is made in a clean and any environment.

Reply:
That sounds like a neat idea.

Reply:
Now that I came back to this, to refresh a bit this PR... is this point covered now that we have the `rpmbuild` directory inside the project's path?

The only thing needed now would be to actually run the build using podman, for which we can add a specific command (e.g., build-podman).

Reply:
I did an attempt with `build-podman`. Please check how it looks, I have tried locally and seems to work fine, but maybe it can be improved or there is something I am missing.

I also tried to add commands to run tests in a container and things like that, but I think I would leave that for future improvements. For the time being, I guess one can just connect to a shell in the container and invoke the script to run the tests.

Reply:
I think adding new target to the make will be more useful instead of a script in contrib.

Since we integrated most of the things in this script, maybe it is time to close this PR and open new one for every feature you want to add (e.g. build vdsm in a container).

Reply:
Having it scripted allows to specify the container image in which you want to build as a script argument. The only way to achieve it in a Makefile is using an envvar (that I know, at least). If being able to select the image is not a desirable feature or using an envvar is ok, I can move it to the Makefile.

Anyway, I still see this script useful, it allows me not to worry about activating the `venv` or adding any other command line options that the script does transparently for you.

I added an option to run as root that deals with the same things as the usual run, but does not overwrite the workdir or the coverage folder. It can be further tuned to clean the new folders up after running if we want.

Also, we could integrate some other scripts in the `contrib` folder into this script, or the other way around. Examples: https://github.com/oVirt/vdsm/blob/master/contrib/shell_helper
https://github.com/oVirt/vdsm/blob/master/contrib/koji_build.sh
these could be updated and integrated together with this script.

Having all them centralized and documented is easier for new contributors.